system/Database/BaseBuilder.php

```diff
@@ -1588,38 +1588,49 @@ public function insertBatch(?array $set = null, ?bool $escape = null, int $batch
 
                 return false; // @codeCoverageIgnore
             }
-        } else {
-            if (empty($set)) {
-                if (CI_DEBUG) {
-                    throw new DatabaseException('insertBatch() called with no data');
-                }
-
-                return false; // @codeCoverageIgnore
+        } elseif (empty($set)) {
+            if (CI_DEBUG) {
+                throw new DatabaseException('insertBatch() called with no data');
             }
 
-            $this->setInsertBatch($set, '', $escape);
+            return false; // @codeCoverageIgnore
         }
 
+        $hasQBSet = ($set === null);
```

Comment:
`()` is not needed as well:

```suggestion
        $hasQBSet = $set === null;
```